src/pages/dashboard/workplace/components/Radar/index.tsx

```diff
@@ -0,0 +1,219 @@
+import { Axis, Chart, Coord, Geom, Tooltip } from 'bizcharts';
+import { Col, Row } from 'antd';
+import React, { Component } from 'react';
+
+import autoHeight from './autoHeight';
+import styles from './index.less';
+
+export interface RadarProps {
+  title?: React.ReactNode;
+  height?: number;
+  padding?: [number, number, number, number];
+  hasLegend?: boolean;
+  data: {
+    name: string;
+    label: string;
+    value: string | number;
+  }[];
+  colors?: string[];
+  animate?: boolean;
+  forceFit?: boolean;
+  tickCount?: number;
+  style?: React.CSSProperties;
+}
+interface RadarState {
+  legendData: {
+    checked: boolean;
+    name: string;
+    color: string;
+    percent: number;
+    value: string;
+  }[];
+}
+/* eslint react/no-danger:0 */
+class Radar extends Component<RadarProps, RadarState> {
+  state: RadarState = {
+    legendData: [],
+  };
+
+  chart: G2.Chart | undefined = undefined;
+
+  node: HTMLDivElement | undefined = undefined;
+
+  componentDidMount() {
+    this.getLegendData();
+  }
+
+  componentDidUpdate(preProps: RadarProps) {
+    const { data } = this.props;
+    if (data !== preProps.data) {
+      this.getLegendData();
+    }
+  }
+
+  getG2Instance = (chart: G2.Chart) => {
+    this.chart = chart;
+  };
+
+  // for custom lengend view
+  getLegendData = () => {
+    if (!this.chart) return;
+    const geom = this.chart.getAllGeoms()[0]; // 获取所有的图形
+    if (!geom) return;
+    const items = (geom as any).get('dataArray') || []; // 获取图形对应的
+
+    const legendData = items.map((item: { color: any; _origin: any }[]) => {
+      // eslint-disable-next-line no-underscore-dangle
+      const origins = item.map(t => t._origin);
+      const result = {
+        name: origins[0].name,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['magic numbers' are not allowed: 0](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)